app/carpool/views.py

```diff
@@ -240,6 +240,10 @@ def modify_ride_request(carpool_id, request_id, action):
             db.session.commit()
             flash("You approved their ride request.")
             # TODO Send email notification to rider
+        elif action == 'cancel':
+            db.session.delete(request)
```

Comment:
what, after a request is denied, user has to cancel the denied request to stop it from showing up on their page? there should def be a way to stop the request from showing up on user's page but this seems like poor UX -- if this is what's actually going on, could you leave a note/todo here or open an issue for later about gracefully resolving denied requests w/o user having to cancel? (request history, maybe?)

Reply:
If the denied request is immediately removed, then the rider that made the request won't have any reference to the denied request on their page.